integration/setup/tenant_cluster.go

```diff
@@ -87,14 +92,57 @@ func EnsureTenantClusterDeleted(ctx context.Context, id string, config Config, w
 	return nil
 }
 
-func crExistsCondition(ctx context.Context, id string, config Config) release.ConditionFunc {
+func crExistsCondition(ctx context.Context, config Config, crd *apiextensionsv1beta1.CustomResourceDefinition, crNamespace, crName string) release.ConditionFunc {
```

Comment:
From signature you can see it's more generic. It needs only `rest.Config` to function. I'm going to move it to `e2e-harness` as a next steps.